Sources/Sentry/include/SentryLegacyProfiler.h

```diff
@@ -0,0 +1,35 @@
+#import "SentryProfilingConditionals.h"
+
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+
+#    import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class SentryId;
+
+@interface SentryLegacyProfiler : NSObject
```

Comment:
`l`: What if this persists until we have another new profiling mode and the Continuous also becomes a legacy? Then we have to name this SentryLegacyLegacyProfiler, and SentryContinuousProfiler becomes SentryLegacyProfiler 😄 . Maybe we should call this `SentryTraceProfiler,` so its name reflects that it requires a traceId to work. But we can also keep it as is.

Reply:
We can do that, although I hope continuous profiling is the last iteration we need :D

Reply:
So I'm going to do this in a separate PR, because I have a few stacked at this point and resolving the conflicts through each will be annoying. https://github.com/getsentry/sentry-cocoa/pull/3952